control-plane/pkg/reconciler/broker/controller.go

```diff
@@ -104,6 +104,22 @@ func NewController(ctx context.Context, watcher configmap.Watcher, configs *Conf
 		Handler:    controller.HandleAll(impl.Enqueue),
 	})
 
+	globalResync := func(_ interface{}) {
+		impl.GlobalResync(brokerInformer.Informer())
+	}
+
+	configmapInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterWithNameAndNamespace(configs.DataPlaneConfigMapNamespace, configs.DataPlaneConfigMapName),
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				globalResync(obj)
```

Comment:
Are you sure we need to global resync on add? Is this going to create a loop when this controller effectively creates the config map?

Reply:
No.

Reply:
AFAIK, AddFunc is called when an object is added to the underlying store.

Reply:
Can we test it?

Reply:
I'm not sure how to test them, do you?
I manually verified that it works as expected.
We need `addFunc` because only one replica will effectively create the CM and we want all replicas to resync the state.

Reply:
Maybe just develop a test that starts the broker, wait for it to be ready, try to send an event, assert that event and then remove the contract config map and try to send and assert another event again... Does it makes sense?

Reply:
Oh, yes, it does, I thought you wanted to catch loops.

Reply:
If that test fails, then there are loops or something else broken, right?

Reply:
Done.